src/model/Patient.ts

```diff
@@ -1,13 +1,51 @@
 import AbstractDBModel from './AbstractDBModel'
+import Name from './Name'
 
 export default class Patient extends AbstractDBModel {
-  firstName: string
+  name: Name
 
-  lastName: string
+  sex: string
 
-  constructor(id: string, rev: string, firstName: string, lastName: string) {
+  dateOfBirth?: string
+
+  isApproximateDateOfBirth?: boolean
+
+  phoneNumber?: string
+
+  email?: string
+
+  address?: string
+
+  preferredLanguage?: string
+
+  occupation?: string
+
+  type?: string
+
+  constructor(
+    id: string,
+    rev: string,
+    name: Name,
```

Comment:
I am not sure we need a custom class for name. Another problem in having 2 depth prop is that we have to check if `.name` exists before doing `.name.given`. I think it is better to flat it.

Reply:
Changed it to flatten the name object.